pkg/runtime/processor/subscriptions.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package processor
+
+import (
+	"context"
+
+	subapi "github.com/dapr/dapr/pkg/apis/subscriptions/v2alpha1"
+	"github.com/dapr/dapr/utils"
+)
+
+func (p *Processor) AddPendingSubscription(ctx context.Context, subscriptions ...subapi.Subscription) bool {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	if p.shutdown.Load() {
+		return false
+	}
+
+	scopedSubs := make([]subapi.Subscription, 0, len(subscriptions))
+	for i, subscription := range subscriptions {
+		if len(subscription.Scopes) > 0 && !utils.Contains[string](subscription.Scopes, p.appID) {
```

Comment:
Can unit tests be added to this method? It contains authorization logic

Reply:
Sure thing- note that security is a mute point here as this is sidecar code so we already have the data. While fine to check again (and have to for backwards compat), we need to add server side filtering for subscriptions [like components](https://github.com/dapr/dapr/pull/7615).

Reply:
Done in https://github.com/dapr/dapr/pull/7583/commits/4c984c4ab68b35cb8fa377bf4d13a568ac031fb7